File: src/paths.zig
/// ~/.local — user-local prefix (XDG_DATA_HOME base)
pub const local_dir = ".local";
/// ~/.local/bin — managed binary directory
pub const bin_dir = "bin";
/// ~/.config — XDG config base directory
pub const config_dir = ".config";
/// ~/.config/dot — dot's own config subdirectory
pub const dot_config_subdir = "dot";
/// Fallback path used when $HOME is not set
pub const fallback_home = "/tmp";
/// Suffix appended during atomic rename (write to .new, then rename over target)
pub const new_file_suffix = ".new";
/// Temporary file extension used during HTTP downloads
pub const tmp_file_ext = ".tmp";

File: src/paths.rs
use crate::error::DotError;
use std::path::PathBuf;

pub const LOCAL_DIR: &str = ".local";
pub const BIN_DIR: &str = "bin";
pub const CONFIG_DIR: &str = ".config";
pub const DOT_CONFIG_SUBDIR: &str = "dot";

pub fn home_dir() -> Result<PathBuf, DotError> {
    dirs::home_dir().ok_or(DotError::NoHome)
}

pub fn dot_config_dir() -> Result<PathBuf, DotError> {
    let home = home_dir()?;
    Ok(home.join(CONFIG_DIR).join(DOT_CONFIG_SUBDIR))
}

pub fn local_bin_dir() -> Result<PathBuf, DotError> {
    let home = home_dir()?;
    Ok(home.join(LOCAL_DIR).join(BIN_DIR))
}

pub fn shell_integration_file(shell: crate::platform::Shell) -> Result<PathBuf, DotError> {
    Ok(local_bin_dir()?.join(shell.integration_file_name()))
}
